prov/sockets/src/sock_fabric.c

```diff
@@ -634,6 +622,57 @@ int sock_node_getinfo(const char *node, const char *service, uint64_t flags,
 	return ret;
 }
 
+static int sock_match_src_addr(struct slist_entry *entry, const void *src_addr)
+{
+	struct sock_host_list_entry *host_entry;
+	host_entry = container_of(entry, struct sock_host_list_entry, entry);
+
+        return (strcmp(host_entry->hostname, (char *) src_addr) == 0);
+}
+
+static int sock_matches_interface(struct slist addr_list, const void *src_addr,
+				   size_t src_len)
+{
+	struct slist_entry *entry;
+	struct addrinfo ai, *rai = NULL;
+	struct sockaddr_in *addr;
+	int ret;
+
+	if (src_len == sizeof(struct sockaddr_in)) {
+		addr = (struct sockaddr_in *)src_addr;
```

Comment:
We shouldn't need a cast here.  You may need to declare addr as const to avoid a warning, and propagate the const through to subcalls.